### ISSUE TITLE
`document.CurrentScript` not accessible from worker

### ISSUE THREAD

**Issue by JeffersGlass** [pyscript/pyscript#1923]
Try the following: 

```py
<script type="py" worker>
    from pyscript import document

    # prints 'None'
    print(document.currentScript) 
</script>
```

It seems `document.currentScript` is not accessible within a worker? Or perhaps I'm misunderstanding how currentScript works.

I stumbled across this when trying to access the `terminal` property of the script, which also fails:
```py
<script type="py" worker terminal>
    from pyscript import document

    #AttributeError: 'NoneType' object has no attribute 'terminal'
    term = document.currentScript.terminal
</script>
```
CC @WebReflection per our Twitter conversation.

**Comment by nasrin1748** [comment]
The following code works for me.

<img width="362" alt="123" src="https://github.com/pyscript/pyscript/assets/79505893/04c6dfa6-86c5-40e5-a2d1-cfd457bfa092">

**Comment by JeffersGlass** [comment]
@nasrin1748 Yes, currentScript works fine in the main thread - it’s specifically when the using the `worker` attribute that there seems to be an issue.

**Comment by WebReflection** [comment]
This is awkward ... I got confused by the `target` export which is supposed to work out of the box in workers too.

True that currentScript is now polluted on main only: https://github.com/pyscript/polyscript/blob/bb0ae0bc90fbe7b6671ecb8202d79194f7d3c382/esm/script-handler.js#L59-L62

This *must be fixed* !!!

**Comment by WebReflection** [comment]
OK, I am confused ... but at least I remembered correctly: https://github.com/pyscript/pyscript/blob/0f788fa2847bb7ac1c7b0346158fda4c61d1635d/pyscript.core/src/plugins-helper.js#L15-L26

Will investigate and fix this soon, thanks again for the report!

**Comment by WebReflection** [comment]
FYI I've fixed (or better, implemented) `currentScript` as part of the *polyscript* logic and module.

This is guaranteed to return the right node even if invoked lazily, which is the opposite of what `document.currentScript` offers, as that doesn't always guarantee the reference is the expected one.

More work needs to be done in PyScript to somehow expose that in either main or worker code, but I start believing we should always use the `polyscript.currentScript` instead of monkey-patching, in an inevitable error prone way, the global `document` reference. I don't think we should necessarily deprecate that, it's still part of the standard, but I think this should be a better way forward:

```python
from pyscript import currentScript
```

We already normalize `window` and `document`, and adding this export would likely never fail expectations and it can be always available in both main and workers thanks to *polyscript* logic behind.

@JeffersGlass @ntoll @madhur-tandon what do you think?

**Comment by WebReflection** [comment]
let's please discuss only here: https://github.com/pyscript/pyscript/discussions/1930#discussioncomment-8094249

**Comment by WebReflection** [comment]
fixed via `__terminal__`

**Comment by nasrin1748** [comment]
So finally what's going to work?
from pyscript import get_current_script
from pyscript import current_script
import pyscript
I am unable to import any of these.

**Comment by WebReflection** [comment]
@nasrin1748 `from polyscript import currentScript` works reliably **only in workers** while there is NO solution on main because a script can be async, it's bootstrap is async too so other screipts might have stolen that reference already, and listeners might run any time in the future and we don't get to catch their execution out of the box and we don't want to override native DOM APIs all over.

In short, `document.currentScript` is problematic on the Web on main and it's as problematic in here too so it won't get fixed on main because it cannot. The `__terminal__` fixes the specific *terminal* case in a consistent way.

**Comment by nasrin1748** [comment]
@WebReflection  i tried importing it in **workers** .Not sure where i am going wrong.
https://_12.pyscriptapps.com/broad-dust/latest/
